WalletWasabi.Tests/UnitTests/PeriodicRunnerTests.cs

```diff
@@ -0,0 +1,87 @@
+#nullable enable
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Bases;
+using Xunit;
+
+namespace WalletWasabi.Tests.UnitTests
+{
+	public class PeriodicRunnerTests
+	{
+		internal class TestRunner : PeriodicRunner
+		{
+			public AutoResetEvent NextRoundAutoResetEvent { get; private set; } = new AutoResetEvent(false);
```

Comment:
We use this [rule](https://documentation.help/StyleCop/SA1201.html) https://github.com/zkSNACKs/WalletWasabi/blob/master/Settings.StyleCop#L477 so properties should be under the ctor, but for whatever reason CodeFactor passed when it shouldn't.

Also this internal class should be placed under the methods, in fact our convention is to have one class per file but maybe it is fine for tests.

Reply:
> We use this [rule](https://documentation.help/StyleCop/SA1201.html) https://github.com/zkSNACKs/WalletWasabi/blob/master/Settings.StyleCop#L477 so properties should be under the ctor, but for whatever reason CodeFactor passed when it shouldn't.

fixed

> Also this internal class should be placed under the methods, in fact our convention is to have one class per file but maybe it is fine for tests.

I'll leave it as it is now. Maybe I'll change it in the future.